src/ice_shelf/MOM_ice_shelf_dynamics.F90

```diff
@@ -706,12 +709,24 @@ subroutine update_ice_shelf(CS, ISS, G, US, time_step, Time, ocean_mass, coupled
      if (CS%id_u_shelf > 0) call post_data(CS%id_u_shelf, CS%u_shelf, CS%diag)
      if (CS%id_v_shelf > 0) call post_data(CS%id_v_shelf, CS%v_shelf, CS%diag)
 !    if (CS%id_t_shelf > 0) call post_data(CS%id_t_shelf,CS%t_shelf,CS%diag)
-     if (CS%id_taudx_shelf > 0) call post_data(CS%id_taudx_shelf, CS%taudx_shelf, CS%diag)
-     if (CS%id_taudy_shelf > 0) call post_data(CS%id_taudy_shelf, CS%taudy_shelf, CS%diag)
+     if (CS%id_taudx_shelf > 0) then
+        taud_x(:,:) = CS%taudx_shelf(:,:)/G%areaT(:,:)
```

Comment:
Rather than dividing by the area here and when calculating the other 3 diagnostics in this change, please consider multiplying by the reciprocal of the area, which is already available as G%IareaT(:,:).  There might be some halo points that are not used and not set to non-zero values, in which case these divisions would lead to NaNs that would cause problems with some compiler settings.  One other way to avoid this is to add explicit do loops and do not do any of these calculations in the unused halo regions.